### DISCUSSION TITLE
audacious refuses to run in qt5 on kubuntu 22.04

### DISCUSSION THREAD

**Discussion by skyprincegamer** [audacious-media-player/audacious#1417] [Q&A] [answered]
```
$ export QT_DEBUG_PLUGINS=1 && audacious
QFactoryLoader::QFactoryLoader() checking directory path "/usr/lib/x86_64-linux-gnu/qt6/plugins/platforms" ...
QFactoryLoader::QFactoryLoader() checking directory path "/usr/bin/platforms" ...
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```

This is most probably audacious is trying to find qt6 stuff which is not installed on my system.
btw gtk mode (audacious -G) works as expected.

**Comment by mschwendt** [comment]
The Qt debug output tells it is Qt 6 looking for its plugins, and the xcb plugin isn't found. In case you've built Audacious yourself, make sure you really enable Qt 5 (although in that case you would also need to install the xcb plugin for Qt).

**Reply by skyprincegamer** [reply]
Thats the thing, I didnt build audacious myself . I had installed it in the past from the Ubuntu repo. When it got updated to the new version , I started seeing this error. How do I tell audacious to use qt5 instead of qt6?

**Reply by mschwendt** [reply]
You can't.

As of Audacious 4.4-beta1 the default is Qt 6 and/or GTK3 (with the option to choose between either one at runtime). Unless someone explicitly builds with Qt 5 instead (and GTK2 as a build option is still available, too).

The xcb plugin package you're missing ought to be added as a strict dependency somewhere. Such a dependency is also needed if using Qt 5.

**Reply by radioactiveman** [reply]
Which Audacious version do you use? And how did you install it? (k)ubuntu 22.04 only ships Audacious 4.1 which does not use Qt 6...

**Reply by skyprincegamer** [reply, marked as answer]
turns out i had installed audacious from the ubuntuhandbook1 ppa (as mentioned on the official audacious website) and not the kubuntu repo . i removed the ubuntuhandbook1 ppa and reinstalled audacious and problem solved. i feel sorry for having wasted people's time

**Reply by radioactiveman** [reply]
> i feel sorry for having wasted people's time

No problem, we should still report this to the PPA provider. Audacious 4.1 is very outdated and other people may want to use the latest version, even with an older Ubuntu.

@PandaJim: Can you reproduce this error on Ubuntu 22.04?

@skyprincegamer: Is `qt6-qpa-plugins` installed on your system? It should contain the xcb plugin which is located at `/usr/lib/x86_64-linux-gnu/qt6/plugins/platforms/libqxcb.so`.

https://packages.ubuntu.com/jammy/amd64/qt6-qpa-plugins/filelist

**Reply by cypherpunk1984** [reply]
> No problem, we should still report this to the PPA provider. Audacious 4.1 is very outdated and other people may want to use the latest version, even with an older Ubuntu.
> 
> @PandaJim: Can you reproduce this error on Ubuntu 22.04?
> 
> @skyprincegamer: Is `qt6-qpa-plugins` installed on your system? It should contain the xcb plugin which is located at `/usr/lib/x86_64-linux-gnu/qt6/plugins/platforms/libqxcb.so`.
> 
> https://packages.ubuntu.com/jammy/amd64/qt6-qpa-plugins/filelist

I'm not OP, but had the same issue with ubuntuhandbook1 PPA. I'm on Linux Mint 21.3 (Ubuntu 22.04) and your answer solved it. After not being able to run audacious, I did 'sudo apt install qt6-qpa-plugins', and now it works perfectly. Thank you so much, I love Audacious! Best player ever!
It seems like the PPA guy didn't put that package as a dependancy. Does anyone have a launchpad account to report this to the guy? This is the repo: https://launchpad.net/~ubuntuhandbook1/+archive/ubuntu/apps

**Reply by PandaJim** [reply]
> > i feel sorry for having wasted people's time
> 
> No problem, we should still report this to the PPA provider. Audacious 4.1 is very outdated and other people may want to use the latest version, even with an older Ubuntu.
> 
> @PandaJim: Can you reproduce this error on Ubuntu 22.04?
> 
> @skyprincegamer: Is `qt6-qpa-plugins` installed on your system? It should contain the xcb plugin which is located at `/usr/lib/x86_64-linux-gnu/qt6/plugins/platforms/libqxcb.so`.
> 
> https://packages.ubuntu.com/jammy/amd64/qt6-qpa-plugins/filelist


Sorry, didn't run into the issue because I've `qt6-qpa-plugins` installed in my system before Audacious 4.4.

The issue seems only affect Ubuntu 22.04, as higher Ubuntu versions built the `libqxcb.so` library into `libqt6gui6` package which should be the correct place. And, I've updated the PPA to include the `qt6-qpa-plugins` as dependency for 22.04.

BTW, **is there a way to set the default audio output plugin** while building process? Audacious 4.4 (qt6 build) in 22.04 in my case defaults to PipeWire while PulseAudio should be default. It causes error popup until switch to PulseAudio in settings.

**Reply by radioactiveman** [reply]
Great, thank you @PandaJim.

> BTW, is there a way to set the default audio output plugin while building process? Audacious 4.4 (qt6 build) in 22.04 in my case defaults to PipeWire while PulseAudio should be default. It causes error popup until switch to PulseAudio in settings.

I would suggest to report an issue with more information for us. My expectation is that PipeWire should also work on 22.04.
Specifying the default output plugin for a build is not possible. Of course you could disable the plugin with Meson or `./configure`.
